sqflite/ios/Classes/SqflitePlugin.m

```diff
@@ -587,7 +587,7 @@ - (void)handleOpenDatabaseCall:(FlutterMethodCall*)call result:(FlutterResult)re
             if (_log) {
                 NSLog(@"Creating parent dir %@", parentDir);
             }
-            [[NSFileManager defaultManager] createDirectoryAtPath:parentDir withIntermediateDirectories:YES attributes:nil error:&error];
+            [[NSFileManager defaultManager] createDirectoryAtPath:parentDir withIntermediateDirectories:YES attributes:@{NSFilePosixPermissions: NSFileProtectionNone} error:&error];
```

Comment:
Maybe some people would prefer NSFileProtectionCompleteUntilFirstUserAuthentication instead of NSFileProtectionNone. How could we allow the developer to choose?

Reply:
Yes, you are right in this regard. Perhaps there can be a flag to enable/disable with some documentation added to readme.